src/components/datepicker/js/calendar.js

```diff
@@ -197,12 +197,24 @@
 
     var boundKeyHandler = angular.bind(this, this.handleKeyEvent);
 
+
+    var handleKeyElement = this.$element;
+
+    // If use the md-calendar directly in the body without datepicker,
+    // handleKeyEvent will disable other inputs on the page.
+    // So only apply the handleKeyEvent on the body when the md-calendar inside datepicker,
+    // otherwise apply on the calendar element only.
+
+    if (this.$element.parent().hasClass('md-datepicker-calendar')) {
+      handleKeyElement = angular.element(document.body);
```

Comment:
It might be a little more readable if you defined the `handleKeyEvent` in an `else` block. Also you can refer to `this.$element` as just `$element`:

```js
if ($element.parent().hasClass('md-datepicker-calendar')) {
  handleKeyElement = angular.element(document.body);
} else {
  handleKeyElement = $element;
}
```